test/charms/test_main/lib/charm.py

```diff
@@ -5,6 +5,7 @@
 import pickle
 
 from op.charm import CharmBase
+from op.main import main
```

Comment:
The value of this test is having a realistic setup that will catch errors real charms will struggle with, and perhaps I'm missing something but it's not clear to me how this would work. Who adds the path of _op_ into _sys.path_ for the statements above to work?

Reply:
It would either be installed as a library (my preference) or side-loaded into the `lib/` directory as it is now.  Either way, with the symlinks pointing to `lib/charm.py` rather than `lib/op/main.py`, it will work without `sys.path` manipulation.  The reason we needed the manipulation before was because of the extra directory layer in `lib/op/main.py`.

The `test/test_main.py` test does cover this, since it uses subprocess and calls the hook.